sklearn/utils/_mocking.py

```diff
@@ -147,9 +147,13 @@ def _check_X_y(self, X, y=None, should_be_fitted=True):
         Parameters
         ----------
         X : array-like of shape (n_samples, n_features)
-            The data set.
+            The data set. Required argument.
         y : array-like of shape (n_samples), default=None
-            The corresponding target, by default None.
+            The corresponding target, by default `None`.
+            If `y = None`, only X is validated. To check y,
+            the checking function, `check_y`, needs to be
+            initiated.
```

Comment:
```suggestion
            The corresponding target, by default `None`.
            `y` is checked only if `check_y=True`.
```

Reply:
We can add the same info in `check_X`.

Reply:
Adressed in https://github.com/scikit-learn/scikit-learn/pull/25997/commits/bf778271bac9af82a1446872ece443108fb2cb18.